### PR TITLE
Org name changed, correcting broken links

### PR DESCRIPTION
**Pull request by lauraporta**
## Description

**What is this PR**

- [x] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**Why is this PR needed?**
I renamed the RSE good practices org

**What does this PR do?**
Changes the name of the org in a link

## References
See conversation on [this PR](https://github.com/neuroinformatics-unit/course-software-good-practice/pull/21)

## Checklist:

- [ ] The code has been tested locally
- [ ] Tests have been added to cover all new functionality
- [ ] The documentation has been updated to reflect any changes
- [ ] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

File: docs/source/courses/collaborative-coding.md [hunk 1]
```diff
@@ -17,10 +17,10 @@ Students should have some initial experience with both Python and version contro
 Students should bring their own laptop with Python installed. If you require any assistance, please contact
 <a href="mailto:adam.tyson@ucl.ac.uk?subject=SWC/GCNU Software Skills">Adam Tyson</a> in advance of the course.
 
-They should also follow [these setup instructions](https://github.com/rse-best-practices-course-2023/rse-best-practices-playground/issues/1) in advance of the course.
+They should also follow [these setup instructions](https://github.com/rse-best-practices-course/rse-best-practices-playground/issues/1) in advance of the course.
```
Comment (alessandrofelder) [inline docs/source/courses/collaborative-coding.md:20]: The problem now (maybe to be tackled separately), is that we might want to include the date in the exercise repo, a this now points to an archived one?

```suggestion
They should also follow [these setup instructions](https://github.com/rse-best-practices-course/rse-best-practices-playground-2024/issues/1) in advance of the course.
```

**Review by alessandrofelder — APPROVED** [review]

File: docs/source/courses/collaborative-coding.md (hunk 1 above)
Comment (alessandrofelder) [reply docs/source/courses/collaborative-coding.md:20]: Or we entirely delete the archived repo from last year (but the student's might use it as a reference, and their forks will continue to exist?)

Comment (lauraporta) [reply docs/source/courses/collaborative-coding.md:20]: Actually, this made me think that we do not want them to follow those setup instructions anymore, so actually this reference is outdated now 😱 shall we delete it directly?

Comment (alessandrofelder) [reply docs/source/courses/collaborative-coding.md:20]: Yea, I think so. We should mention that they need Git and Github setup too? We can link them to https://neuroinformatics.dev/slides-SWC-PhD-intro/#/section

Comment (lauraporta) [reply docs/source/courses/collaborative-coding.md:20]:
```suggestion
They should also follow [these setup instructions](https://neuroinformatics.dev/slides-SWC-PhD-intro/#/section) in advance of the course.
```

Comment (lauraporta) [reply docs/source/courses/collaborative-coding.md:20]: If this is OK to you I commit my latest suggestion and merge

Comment (alessandrofelder) [reply docs/source/courses/collaborative-coding.md:20]: great! Yes, this is OK with me!

**Pull request merged**